Cannot print reference to string-alias in struct (builder error)

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
```
$ v up
Updating V...
Current V version:
V 0.2.2 c03798e, timestamp: 2021-02-26 00:28:47 +0100
```

**OS:** 
```
$ v doctor
OS: linux, Ubuntu 20.04 LTS (WSL 2)
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
CC version: cc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
... 
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 639061b.c03798e

Git version: git version 2.25.1
Git vroot status: weekly.2021.7-171-gc03798e3
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81
```

**What did you do?**
Execute this program:
```
type SS = string
struct ST {
        data &SS
}
fn main(){
        mut val := ST{
                data: &SS("hi")
        }
        println(val.data)
}
```

**What did you expect to see?**
print &hi or some string to console.
 
**What did you see instead?**
```
$ v -stats -cg -showcc ./exp/try.v 
checker summary: 0 V errors, 0 V warnings
generated C source code size: 8713 lines, 322769 bytes

=====================
> C compiler cmd: /home/youyuan/code/v/thirdparty/tcc/tcc.exe "@/tmp/v/try.17391407795664741950.tmp.c.rsp"
> C compiler response file /tmp/v/try.17391407795664741950.tmp.c.rsp:
  -std=gnu99 -g -o "/home/youyuan/code/mongo/exp/try" "/tmp/v/try.17391407795664741950.tmp.c" -bt25 -rdynamic -lm -lpthread -ldl 
=====================

=====================
> C compiler cmd: cc "@/tmp/v/try.17391407795664741950.tmp.c.rsp"
> C compiler response file /tmp/v/try.17391407795664741950.tmp.c.rsp:
  -std=gnu99 -g -o "***" "/tmp/v/try.17391407795664741950.tmp.c" -rdynamic -lm -lpthread -ldl 
=====================
/tmp/v/try.17391407795664741950.tmp.c:8622: error: cannot convert 'struct string' to 'struct string *'
builder error: 
```

generated C code snip that has problem:
```c
void main__main(void) {
	main__ST val = (main__ST){.data = ((_SLIT("hi"))),};
	println(val.data);
                   //~~argument of type "main__SS *" is incompatible with parameter of type "string"C/C++(167) 
}
```
Error goes away when modified to this:
```c
void main__main(void) {
	main__ST val = (main__ST){.data = &((_SLIT("hi"))),};
	println(*val.data);         
}
```

--- a/vlib/v/gen/c/auto_str_methods.v
+++ b/vlib/v/gen/c/auto_str_methods.v
@@ -916,7 +916,8 @@ fn (mut g Gen) gen_str_for_struct(info ast.Struct, styp string, str_fn_name stri
 			funcprefix += 'isnil(it.${c_name(field.name)})'
 			funcprefix += ' ? _SLIT("nil") : '
 			// struct, floats and ints have a special case through the _str function
-			if sym.kind != .struct_ && !field.typ.is_int_valptr() && !field.typ.is_float_valptr() {
+			if sym.kind !in [.struct_, .alias] && !field.typ.is_int_valptr()
+				&& !field.typ.is_float_valptr() {
 				funcprefix += '*'
 			}
 		}

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5519,6 +5519,9 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 							|| sfield.typ.is_pointer()) && !sfield.typ.is_number() {
 							g.write('/* autoref */&')
 						}
+						if sfield.typ.is_ptr() && field_type_sym.kind == .alias {
+							g.write('&')
+						}
 						g.expr_with_cast(sfield.expr, sfield.typ, sfield.expected_type)
 					}
 				}

--- a/vlib/v/gen/c/str.v
+++ b/vlib/v/gen/c/str.v
@@ -85,6 +85,9 @@ fn (mut g Gen) gen_expr_to_string(expr ast.Expr, etype ast.Type) {
 		g.expr(expr)
 		g.write(')')
 	} else if typ == ast.string_type {
+		if etype.is_ptr() {
+			g.write('*')
+		}
 		g.expr(expr)
 	} else if typ == ast.bool_type {
 		g.expr(expr)

--- a/vlib/v/tests/struct_with_reference_alias_fields.v
+++ b/vlib/v/tests/struct_with_reference_alias_fields.v
@@ -1,0 +1,13 @@
+type SS = string
+
+struct ST {
+	data &SS
+}
+
+fn test_struct_with_reference_alias_fields() {
+	mut val := ST{
+		data: &SS('hi')
+	}
+	println(val.data)
+	assert '$val.data' == 'hi'
+}

